pathgen/pathgen.go

```diff
@@ -767,45 +775,76 @@ func getNodeDataMap(ir *ygen.IR, fakeRootName, schemaStructPkgAccessor, pathStru
 				localGoTypeName = mType.NativeType
 			case field.Type == ygen.LeafListNode:
 				goTypeName = "[]" + mType.NativeType
+				localGoTypeName = goTypeName
 			default:
 				goTypeName = mType.NativeType
-			}
-			if localGoTypeName == "" {
-				localGoTypeName = goTypeName
+				if localGoTypeName == "" {
+					localGoTypeName = goTypeName
+				}
 			}
 
-			var yangTypeName string
-			if field.Flags != nil {
-				yangTypeName = field.Flags[yangTypeNameFlagKey]
-			}
-			nodeDataMap[pathStructName] = &NodeData{
+			nodeData := NodeData{
 				GoTypeName:            goTypeName,
 				LocalGoTypeName:       localGoTypeName,
 				GoFieldName:           goFieldNameMap[fieldName],
 				SubsumingGoStructName: subsumingGoStructName,
 				IsLeaf:                isLeaf,
 				IsScalarField:         gogen.IsScalarField(field),
 				HasDefault:            isLeaf && (len(field.YANGDetails.Defaults) > 0 || mType.DefaultValue != nil),
-				YANGTypeName:          yangTypeName,
-				YANGPath:              field.YANGDetails.Path,
-				GoPathPackageName:     goPackageName(field.YANGDetails.RootElementModule, splitByModule, false, packageName, trimPrefix, packageSuffix),
-				DirectoryName:         field.YANGDetails.Path,
+				YANGTypeName: func() string {
+					if field.Flags != nil {
+						return field.Flags[yangTypeNameFlagKey]
+					} else {
+						return ""
+					}
+				}(),
+				YANGPath:          field.YANGDetails.Path,
+				GoPathPackageName: goPackageName(field.YANGDetails.RootElementModule, splitByModule, false, packageName, trimPrefix, packageSuffix),
+				DirectoryName:     field.YANGDetails.Path,
 			}
 			switch {
-			case isLeaf:
-				nodeDataMap[pathStructName].YANGFieldName = fieldName
-				nodeDataMap[pathStructName].DirectoryName = dir.Path
-			case !isLeaf && field.YANGDetails.OrderedByUser: // "ordered-by user" lists
+			case !isLeaf && field.Type == ygen.ListNode && len(fieldDir.ListKeys) > 0 && !(fieldDir.CompressedTelemetryAtomic || field.YANGDetails.OrderedByUser):
```

Comment:
comment these cases or simplify the conditions

Reply:
Added a helper to improve readability